index.js

```diff
@@ -90,67 +146,59 @@ function triggerInstalls(approvedUpdates) {
-        state.updates = require('exclude')(state.updates, msg.detail);
-        button.badge = state.updates.length;
-        triggerInstalls(msg.detail);
-        break;
-      case 'loaded':
-        sendToWeb({ type: 'addon-available' });
-        break;
-      default:
-        sendToWeb({ type: 'echo', data: msg });
-        break;
-    }
-  });
-
-  const installListeners = {
-    onInstallEnded: function(install, addon) {
-      sendToWeb({type: 'addon-install:install-ended', detail: formatInstallData(install, addon)});
-    },
-    onInstallFailed: function(install) {
-      sendToWeb({type: 'addon-install:install-failed', detail: formatInstallData(install)});
-    },
-    onInstallStarted: function(install) {
-      sendToWeb({type: 'addon-install:install-started', detail: formatInstallData(install)});
-    },
-    onNewInstall: function(install) {
-      sendToWeb({type: 'addon-install:install-new', detail: formatInstallData(install)});
-    },
-    onInstallCancelled: function(install) {
-      sendToWeb({type: 'addon-install:install-cancelled', detail: formatInstallData(install)});
-    },
-    onDownloadStarted: function(install) {
-      sendToWeb({type: 'addon-install:download-started', detail: formatInstallData(install)});
-    },
-    onDownloadProgress: function(install) {
-      sendToWeb({type: 'addon-install:download-progress', detail: formatInstallData(install)});
-    },
-    onDownloadEnded: function(install) {
-      sendToWeb({type: 'addon-install:download-ended', detail: formatInstallData(install)});
-    },
-    onDownloadCancelled: function(install) {
-      sendToWeb({type: 'addon-install:download-cancelled', detail: formatInstallData(install)});
-    },
-    onDownloadFailed: function(install) {
-      sendToWeb({type: 'addon-install:download-failed', detail: formatInstallData(install)});
-    }
+function formatInstallData(install, addon) {
+  let formatted = {
+    'name': install.name ? install.name : '',
+    'error': install.error,
+    'state': install.state,
+    'version': install.version ? install.version : '',
+    'progress': install.progress,
+    'maxProgress': install.maxProgress
   };
 
-  AddonManager.addInstallListener(installListeners);
+  if (addon) {
+    formatted = require('xtend')(formatted, {
+      'description': addon.description,
+      'homepageURL': addon.homepageURL,
+      'iconURL': addon.iconURL,
+      'size': addon.size,
+      'signedState': addon.signedState,
+      'permissions': addon.permissions
+    });
+  }
+
+  return formatted;
 }
+
+AddonManager.addInstallListener({
+  onInstallEnded: function(install, addon) {
+    app.send('addon-install:install-ended', formatInstallData(install, addon));
+  },
+  onInstallFailed: function(install) {
+    app.send('addon-install:install-failed', formatInstallData(install));
+  },
+  onInstallStarted: function(install) {
+    app.send('addon-install:install-started', formatInstallData(install));
+  },
+  onNewInstall: function(install) {
```

Comment:
:eyes: 

Looks like this is the only one that breaks the "on[Download | Install]Thing" rules. Not sure if you want to rename to the less eye-pleasing, but more consistent `onInstallNew`. Or not, whatever.